tests/integration/backward_compatible/proxy/reliable_topic_test.py

```diff
@@ -0,0 +1,448 @@
+            },
+            "block": {
+                "overload_policy": TopicOverloadPolicy.BLOCK,
+            },
+            "error": {
+                "overload_policy": TopicOverloadPolicy.ERROR,
+            },
+        }
+        return config
+
+    def setUp(self):
+        self.topics = []
+        self.topic = self.get_topic(random_string())
+
+    def tearDown(self):
+        for topic in self.topics:
+            topic.destroy()
+
+    def test_add_listener_with_function(self):
+        topic = self.get_topic(random_string())
+
+        collector = event_collector()
+        registration_id = topic.add_listener(collector)
+        self.assertIsNotNone(registration_id)
+
+        topic.publish("a")
+        topic.publish("b")
+
+        self.assertTrueEventually(
+            lambda: self.assertEqual(["a", "b"], list(map(lambda m: m.message, collector.events)))
+        )
+
+    def test_add_listener(self):
+        topic = self.get_topic(random_string())
+
+        messages = []
+
+        class Listener(ReliableMessageListener):
+            def on_message(self, message):
+                messages.append(message.message)
+
+            def retrieve_initial_sequence(self):
+                return -1
+
+            def store_sequence(self, sequence):
+                pass
+
+            def is_loss_tolerant(self):
+                return False
+
+            def is_terminal(self, error):
+                return False
+
+        registration_id = topic.add_listener(Listener())
+        self.assertIsNotNone(registration_id)
+
+        topic.publish("a")
+        topic.publish("b")
+
+        self.assertTrueEventually(lambda: self.assertEqual(["a", "b"], messages))
+
+    def test_add_listener_with_retrieve_initial_sequence(self):
+        topic = self.get_topic(random_string())
+
+        messages = []
+
+        class Listener(ReliableMessageListener):
+            def on_message(self, message):
+                messages.append(message.message)
+
+            def retrieve_initial_sequence(self):
+                return 5
+
+            def store_sequence(self, sequence):
+                pass
+
+            def is_loss_tolerant(self):
+                return False
+
+            def is_terminal(self, error):
+                return False
+
+        registration_id = topic.add_listener(Listener())
+        self.assertIsNotNone(registration_id)
+
+        topic.publish_all(range(10))
+
+        self.assertTrueEventually(lambda: self.assertEqual(list(range(5, 10)), messages))
+
+    def test_add_listener_with_store_sequence(self):
+        topic = self.get_topic(random_string())
+
+        sequences = []
+
+        class Listener(ReliableMessageListener):
+            def on_message(self, message):
+                pass
+
+            def retrieve_initial_sequence(self):
+                return -1
+
+            def store_sequence(self, sequence):
+                sequences.append(sequence)
```

Comment:
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/test/java/com/hazelcast/topic/impl/reliable/DurableSubscriptionTest.java

Can you add a test like this to see store and retrieve is working correctly.

Reply:
Added the same test